upstairs/src/lib.rs

```diff
@@ -1010,6 +1018,11 @@ pub struct Upstairs {
      */
     ds_state: Mutex<Vec<DsState>>,
 
+    /*
+     * UUID for each downstairs, index by client ID
```

Comment:
the RFD mentions:

> Nexus sends to each Crucible Agent the size of the segment along with UUIDs describing both the overall storage volume and a UUID for this specific downstairs.

but I don't see a distinction in this PR between the volume UUID and downstairs UUID. We might even want different types for that so there won't be confusion in the code.

Reply:
For sure there will be a pool wide UUID.  There will possibly be certificates or keys or other bits of info that the upstairs will have.  From the  downstairs point of view, I'm not sure if it needs to know the UUID of the overall pool, but if it does we can add it (along with the rest) when we decide what it should have.

I'm still considering making some struct that holds a bunch of information about each downstairs, but as I don't know know exactly what we need yet, I'm holding off on that.

For this change, I'm just handling the upstairs/downstairs UUID passing.  There is a bunch more work here, but I wanted to prevent this review from turning into a monster, and this seemed like a good place to pause.  I'm already a few days into the next part of this, so, don't worry, more is coming :)

Reply:
:+1: